src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestBuiltInIntrinsics1.ll

```diff
@@ -49,24 +49,24 @@ entry:
   %27 = getelementptr inbounds { %String* }, { %String* }* %26, i32 0, i32 0
   store %String* %24, %String** %27, align 8
   call void @__quantum__rt__callable_invoke(%Callable* %19, %Tuple* %25, %Tuple* null)
```

Comment:
I'm not sure if this was the incomplete part you referred to in the description or if I'm missing something about these changes, but I would've expected all of the calls to `__quantum__rt__callable_invoke` to have been replaced by the calls to the global functions the callable's correlate to, since none of these callables utilize capture tuples or escape their scope. But for some reason only the last invocation was replaced. Is this known unfinished work or a bug?

Reply:
Oh, never mind, I can see it now. The `%options` value is a UDT tuple that is created via `__quantum__rt__tuple_copy` so all of that value's cached state before the last copy-update that defines it is lost, so only the last callable value is known by the compiler. The logic as written would work for all three callables if callabes were completely local and stack allocated, which would happen more reliably if this were targeted QIR that went through the inlining pass and stack allocated tuples. Fascinating.

Reply:
if I skip the UDT and just use a tuple like this:
```qsharp
        let options = (Message, DumpMachine, DumpMachine);
        let (a, b, c) = options;
        a("foo");
        b("bar");
        c();
```
Then I can see all three functions being replaced!

Reply:
Yes, exactly; this change really is primarily intended for targeted QIR - but it is enabled in general (didn't see a reason why it shouldn't be).